netbox/operations.py

```diff
@@ -0,0 +1,428 @@
+"""
+Copyright start
+MIT License
+Copyright (c) 2023 Fortinet Inc
+Copyright end
+"""
+
+import requests
+from connectors.core.connector import get_logger, ConnectorError
+
+logger = get_logger('netbox')
+
+
+class Netbox:
+    def __init__(self, config):
+        self.server_url = config.get('server_url')
+        if not self.server_url.startswith('https://'):
+            self.server_url = 'https://' + self.server_url
+        self.server_url.strip('/')
+        self.api_key = config.get('api_key')
+        self.verify_ssl = config.get('verify_ssl')
+
+    def make_request(self, endpoint, method='get', data=None, params=None, files=None):
+        try:
+            url = self.server_url + endpoint
+            logger.info('Executing url {}'.format(url))
+            headers = {'Authorization': f"Token {self.api_key}", 'Content-Type': 'application/json'}
+
+            # # CURL UTILS CODE
+            # try:
+            #     from connectors.debug_utils.curl_script import make_curl
+            #     make_curl(method, endpoint, headers=headers, params=params, data=data, verify_ssl=self.verify_ssl)
+            # except Exception as err:
+            #     logger.error(f"Error in curl utils: {str(err)}")
+
+            response = requests.request(method, url, params=params, files=files, data=data, headers=headers,
+                                        verify=self.verify_ssl)
+            if response.ok:
+                logger.info('successfully get response for url {}'.format(url))
+                if method == 'delete':
```

Comment:
Please check case of the string 'delete' or check with method.lower() == 'delete'